packages/mui-material/src/MobileStepper/MobileStepper.test.js

```diff
@@ -111,4 +111,31 @@ describe('<MobileStepper />', () => {
     rerender(<MobileStepper {...defaultProps} variant="progress" steps={3} activeStep={2} />);
     expect(screen.getByRole('progressbar').getAttribute('aria-valuenow')).to.equal('100');
   });
+
+  it('should set value correctly when steps is set to 1', () => {
+    const { getByRole } = render(<MobileStepper {...defaultProps} variant="progress" steps={1} />);
+    expect(screen.getByRole('progressbar').getAttribute('aria-valuenow')).to.equal('100');
+    fireEvent.click(getByRole('button', { name: 'next' }));
+    expect(screen.getByRole('progressbar').getAttribute('aria-valuenow')).to.equal('100');
+    fireEvent.click(getByRole('button', { name: 'back' }));
+    expect(screen.getByRole('progressbar').getAttribute('aria-valuenow')).to.equal('100');
+  });
+
+  it('should set value correctly when steps is updated with 1 & 2', () => {
+    const { rerender } = render(<MobileStepper {...defaultProps} variant="progress" steps={1} />);
+    expect(screen.getByRole('progressbar').getAttribute('aria-valuenow')).to.equal('100');
+    rerender(<MobileStepper {...defaultProps} variant="progress" steps={2} />);
+    expect(screen.getByRole('progressbar').getAttribute('aria-valuenow')).to.equal('0');
+    rerender(<MobileStepper {...defaultProps} variant="progress" steps={1} />);
+    expect(screen.getByRole('progressbar').getAttribute('aria-valuenow')).to.equal('100');
+  });
+
+  it('should set value correctly when active step is greater than steps', () => {
+    const { rerender } = render(
+      <MobileStepper {...defaultProps} variant="progress" activeStep={5} steps={6} />,
+    );
+    expect(screen.getByRole('progressbar').getAttribute('aria-valuenow')).to.equal('100');
+    rerender(<MobileStepper {...defaultProps} variant="progress" activeStep={5} steps={3} />);
```

Comment:
Now after looking at this test, it does not look right that the active step should be greater than the steps. When the steps is changed to a lower value after changing the state, the active step should also be updated by the developer using state to a lower value. This should be handled by the developer not by us. I think we should remove [this](https://github.com/mui/material-ui/pull/37079/files#diff-229a6eee040d2614dd5fc2d0860c68420f4cc0b6b81ee84b4d95c6f16f0d36f3R118) logic.

Reply:
This has been implemented @ZeeshanTamboli